drivers/ethernet/dwc_xgmac/eth_dwc_xgmac.c

```diff
@@ -0,0 +1,1929 @@
+
+	caps = (ETHERNET_LINK_1000BASE_T | ETHERNET_LINK_100BASE_T | ETHERNET_LINK_10BASE_T |
+		ETHERNET_AUTO_NEGOTIATION_SET | ETHERNET_DUPLEX_SET);
+
+#ifdef CONFIG_ETH_DWC_XGMAC_RX_CS_OFFLOAD
+	caps |= ETHERNET_HW_RX_CHKSUM_OFFLOAD;
+#endif
+
+#ifdef CONFIG_ETH_DWC_XGMAC_TX_CS_OFFLOAD
+	caps |= ETHERNET_HW_TX_CHKSUM_OFFLOAD;
+#endif
+
+#if (!CONFIG_ETH_DWC_XGMAC_PROMISCUOUS_EXCEPTION && CONFIG_NET_PROMISCUOUS_MODE)
+	caps |= ETHERNET_PROMISC_MODE;
+#endif
+
+#ifdef CONFIG_ETH_DWC_XGMAC_HW_FILTERING
+	caps |= ETHERNET_HW_FILTERING;
+#endif
+
+	return caps;
+}
+
+#if defined(CONFIG_NET_STATISTICS_ETHERNET)
+/**
+ * @brief XGMAC statistics data request function
+ * Returns a pointer to the statistics data of the current XGMAC controller.
+ *
+ * @param dev Pointer to the ethernet device
+ * @return Pointer to the current XGMAC device's statistics data
+ */
+static struct net_stats_eth *eth_dwc_xgmac_stats(const struct device *dev)
+{
+	struct eth_dwc_xgmac_dev_data *dev_data = (struct eth_dwc_xgmac_dev_data *)dev->data;
+
+	return &dev_data->stats;
+}
+#endif
+
+static const struct ethernet_api eth_dwc_xgmac_apis = {
+	.iface_api.init = eth_dwc_xgmac_iface_init,
+	.send = eth_dwc_xgmac_send,
+	.start = eth_dwc_xgmac_start_device,
+	.stop = eth_dwc_xgmac_stop_device,
+	.get_capabilities = eth_dwc_xgmac_get_capabilities,
+	.set_config = eth_dwc_xgmac_set_config,
+	.get_config = eth_dwc_xgmac_get_config,
+#ifdef CONFIG_NET_STATISTICS_ETHERNET
+	.get_stats = eth_dwc_xgmac_stats,
+#endif /* CONFIG_NET_STATISTICS_ETHERNET */
+};
+
+/* Interrupt configuration function macro */
+#define ETH_DWC_XGMAC_CONFIG_IRQ_FUNC(port)                                                        \
+	static void eth_dwc_xgmac##port##_irq_config(const struct device *dev)                     \
+	{                                                                                          \
+		ARG_UNUSED(dev);                                                                   \
+		IRQ_CONNECT(DT_INST_IRQN(port), DT_INST_IRQ(port, priority), eth_dwc_xgmac_isr,    \
+			    DEVICE_DT_INST_GET(port), 0);                                          \
+	}                                                                                          \
+	static void eth_dwc_xgmac##port##_irq_enable(const struct device *dev, bool en)            \
+	{                                                                                          \
+		ARG_UNUSED(dev);                                                                   \
+		en ? irq_enable(DT_INST_IRQN(port)) : irq_disable(DT_INST_IRQN(port));             \
+	}                                                                                          \
+	volatile uint32_t eth_dwc_xgmac##port##_dma_ch_int_status[DT_INST_PROP(port, num_dma_ch)];
+
+#define ETH_DWC_XGMAC_ALLOC_DMA_DESC(port)                                                         \
+	mem_addr_t eth_dwc_xgmac##port##_tx_pkts[DT_INST_PROP(port, num_dma_ch)]                   \
```

Comment:
suggestion,
you have many macros as per my view. another 2 more will likely increase readability.
#define CHLCNT(n) DT_INST_PROP(port, num_dma_ch)
#define MAX_TX_RING(n) DT_INST_PROP(port, dma_ch_tdrl)

and use like this, 
mem_addr_t eth_dwc_xgmac##port##_tx_pkts[ CHLCNT(port) ][ MAX_TX_RING(port) ];